### ISSUE TITLE
Cannot install python-fcl on pop os

### ISSUE THREAD

**Issue by hamdi-99** [BerkeleyAutomation/python-fcl#31]
I am facing this issue while installing the python-fcl package 
(I have octomap installed ) i followed the steps and always get this error message :
Collecting python-fcl
  Using cached python-fcl-0.0.12.tar.gz (17 kB)
Requirement already satisfied: numpy in /home/hamdiharaketi/anaconda3/envs/downgrade/lib/python3.8/site-packages (from python-fcl) (1.20.1)
Requirement already satisfied: cython in /home/hamdiharaketi/anaconda3/envs/downgrade/lib/python3.8/site-packages (from python-fcl) (0.29.23)
Building wheels for collected packages: python-fcl
  Building wheel for python-fcl (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/hamdiharaketi/anaconda3/envs/downgrade/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-0kgk9pn3/python-fcl_c4c6e2e8fc484e71b7203662422d1846/setup.py'"'"'; __file__='"'"'/tmp/pip-install-0kgk9pn3/python-fcl_c4c6e2e8fc484e71b7203662422d1846/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-w9m44af_
       cwd: /tmp/pip-install-0kgk9pn3/python-fcl_c4c6e2e8fc484e71b7203662422d1846/
  Complete output (28 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.8
  creating build/lib.linux-x86_64-3.8/fcl
  copying fcl/collision_data.py -> build/lib.linux-x86_64-3.8/fcl
  copying fcl/version.py -> build/lib.linux-x86_64-3.8/fcl
  copying fcl/__init__.py -> build/lib.linux-x86_64-3.8/fcl
  running build_ext
  cythoning fcl/fcl.pyx to fcl/fcl.cpp
  building 'fcl.fcl' extension
  creating build/temp.linux-x86_64-3.8
  creating build/temp.linux-x86_64-3.8/fcl
  gcc -pthread -B /home/hamdiharaketi/anaconda3/envs/downgrade/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/usr/include -I/usr/local/include -I/usr/include/eigen3 -I/home/hamdiharaketi/anaconda3/envs/downgrade/lib/python3.8/site-packages/numpy/core/include -I/home/hamdiharaketi/anaconda3/envs/downgrade/include/python3.8 -c fcl/fcl.cpp -o build/temp.linux-x86_64-3.8/fcl/fcl.o -std=c++11
  cc1plus: attention: l'option de la ligne de commande « -Wstrict-prototypes » est valable pour C/ObjC mais pas pour C++
  Dans le fichier inclus depuis /home/hamdiharaketi/anaconda3/envs/downgrade/lib/python3.8/site-packages/numpy/core/include/numpy/ndarraytypes.h:1944,
                   depuis /home/hamdiharaketi/anaconda3/envs/downgrade/lib/python3.8/site-packages/numpy/core/include/numpy/ndarrayobject.h:12,
                   depuis /home/hamdiharaketi/anaconda3/envs/downgrade/lib/python3.8/site-packages/numpy/core/include/numpy/arrayobject.h:4,
                   depuis fcl/fcl.cpp:633:
  /home/hamdiharaketi/anaconda3/envs/downgrade/lib/python3.8/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: attention: #warning "Using deprecated NumPy API, disable it with " "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
     17 | #warning "Using deprecated NumPy API, disable it with " \
        |  ^~~~~~~
  fcl/fcl.cpp:656:10: erreur fatale: octomap/OccupancyOcTreeBase.h : Aucun fichier ou dossier de ce type
    656 | #include "octomap/OccupancyOcTreeBase.h"
        |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  compilation terminée.
  error: command 'gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for python-fcl
  Running setup.py clean for python-fcl
Failed to build python-fcl
Installing collected packages: python-fcl
    Running setup.py install for python-fcl ... error
    ERROR: Command errored out with exit status 1:
     command: /home/hamdiharaketi/anaconda3/envs/downgrade/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-0kgk9pn3/python-fcl_c4c6e2e8fc484e71b7203662422d1846/setup.py'"'"'; __file__='"'"'/tmp/pip-install-0kgk9pn3/python-fcl_c4c6e2e8fc484e71b7203662422d1846/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-p3qgen2b/install-record.txt --single-version-externally-managed --compile --install-headers /home/hamdiharaketi/anaconda3/envs/downgrade/include/python3.8/python-fcl
         cwd: /tmp/pip-install-0kgk9pn3/python-fcl_c4c6e2e8fc484e71b7203662422d1846/
    Complete output (28 lines):
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.8
    creating build/lib.linux-x86_64-3.8/fcl
    copying fcl/collision_data.py -> build/lib.linux-x86_64-3.8/fcl
    copying fcl/version.py -> build/lib.linux-x86_64-3.8/fcl
    copying fcl/__init__.py -> build/lib.linux-x86_64-3.8/fcl
    running build_ext
    skipping 'fcl/fcl.cpp' Cython extension (up-to-date)
    building 'fcl.fcl' extension
    creating build/temp.linux-x86_64-3.8
    creating build/temp.linux-x86_64-3.8/fcl
    gcc -pthread -B /home/hamdiharaketi/anaconda3/envs/downgrade/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/usr/include -I/usr/local/include -I/usr/include/eigen3 -I/home/hamdiharaketi/anaconda3/envs/downgrade/lib/python3.8/site-packages/numpy/core/include -I/home/hamdiharaketi/anaconda3/envs/downgrade/include/python3.8 -c fcl/fcl.cpp -o build/temp.linux-x86_64-3.8/fcl/fcl.o -std=c++11
    cc1plus: attention: l'option de la ligne de commande « -Wstrict-prototypes » est valable pour C/ObjC mais pas pour C++
    Dans le fichier inclus depuis /home/hamdiharaketi/anaconda3/envs/downgrade/lib/python3.8/site-packages/numpy/core/include/numpy/ndarraytypes.h:1944,
                     depuis /home/hamdiharaketi/anaconda3/envs/downgrade/lib/python3.8/site-packages/numpy/core/include/numpy/ndarrayobject.h:12,
                     depuis /home/hamdiharaketi/anaconda3/envs/downgrade/lib/python3.8/site-packages/numpy/core/include/numpy/arrayobject.h:4,
                     depuis fcl/fcl.cpp:633:
    /home/hamdiharaketi/anaconda3/envs/downgrade/lib/python3.8/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: attention: #warning "Using deprecated NumPy API, disable it with " "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
       17 | #warning "Using deprecated NumPy API, disable it with " \
          |  ^~~~~~~
    fcl/fcl.cpp:656:10: erreur fatale: octomap/OccupancyOcTreeBase.h : Aucun fichier ou dossier de ce type
      656 | #include "octomap/OccupancyOcTreeBase.h"
          |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    compilation terminée.
    error: command 'gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /home/hamdiharaketi/anaconda3/envs/downgrade/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-0kgk9pn3/python-fcl_c4c6e2e8fc484e71b7203662422d1846/setup.py'"'"'; __file__='"'"'/tmp/pip-install-0kgk9pn3/python-fcl_c4c6e2e8fc484e71b7203662422d1846/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-p3qgen2b/install-record.txt --single-version-externally-managed --compile --install-headers /home/hamdiharaketi/anaconda3/envs/downgrade/include/python3.8/python-fcl Check the logs for full command output.

**Comment by Donglin-Wang2** [comment]
I am facing the same issue on Pop OS. Have you found a solution yet?

**Comment by Donglin-Wang2** [comment]
I think I resolved the issue:
TL;DR:

```
git clone https://github.com/CyrilWaechter/python-fcl.git
pip install ./python-fcl/
```

**NOTE: This works as of 2021/08/31 and would likely change in the future. See the overview for more information**

## Overview
In case that anyone stumbled upon this thread and still have the same issue, I found a fix. The issue is caused by the incompatibility between the python-fcl and the actual fcl package (written in c++). After the fcl updated from version 0.5 to 0.6, there have been some issues. However, I was able to find a pull request #29 in that help mitigated the problem.  

## System Setup
Pop OS
conda environment with python 3.8

**Comment by hamdi-99** [comment]
@Donglin-Wang2 i resolved the issue in another manner, it was a matter of versions so i cloned the latest repo and i created a symbolic link for every library file or folder which is marked as "does not exist"
There is a problem in synchronisation between the updates
